circuits-circom/test/regexes/paylah/paylah_all.spec.ts

```diff
@@ -0,0 +1,170 @@
+
+        assert(Fr.eq(Fr.e(witness[1]), Fr.e(1)));
+    });
+
+    it("Should reveal amount correctly", async () => {
+        const witness = await subject(input);
+
+        const expected = Array(textToAsciiArray("<td>Amount:</td>\r\n<td>SGD").length).fill("0")
+            .concat(textToAsciiArray("300.00"))
+            .concat(textToAsciiArray("</td>\r\n").fill("0"))
+            .concat(textToAsciiArray("</tr>\r\n<tr>\r\n").fill("0"))
+            .concat(textToAsciiArray("<td>From:</td>\r\n").fill("0"))
+            .concat(textToAsciiArray("<td>PayLah! Wallet (Mobile ending 1111)</td>\r\n").fill("0"))
+            .concat(textToAsciiArray("</tr>\r\n<tr>\r\n").fill("0"))
+            .concat(textToAsciiArray("<td>To:</td>\r\n<td>Steven 'O K-M Jr. (Mobile ending 2222)</td>\r\n").fill("0"))
+
+        const result = witness.slice(2, input.msg.length + 2);
+
+        assert.equal(JSON.stringify(result), JSON.stringify(expected), true);
+    });
+
+    it("Should reveal the payer mobile num correctly", async () => {
+        const witness = await subject(input);
+
+        const expected = Array(textToAsciiArray("<td>Amount:</td>\r\n<td>SGD").length).fill("0")
+            .concat(textToAsciiArray("300.00")).fill("0")
+            .concat(textToAsciiArray("</td>\r\n").fill("0"))
+            .concat(textToAsciiArray("</tr>\r\n<tr>\r\n").fill("0"))
+            .concat(textToAsciiArray("<td>From:</td>\r\n").fill("0"))
+            .concat(textToAsciiArray("<td>PayLah! Wallet (Mobile ending ")).fill("0")
+            .concat(textToAsciiArray("1111"))
+            .concat(textToAsciiArray(")</td>\r\n").fill("0"))
+            .concat(textToAsciiArray("</tr>\r\n<tr>\r\n").fill("0"))
+            .concat(textToAsciiArray("<td>To:</td>\r\n<td>Steven 'O K-M Jr. (Mobile ending 2222)</td>\r\n").fill("0"));
+
+        const result = witness.slice(input.msg.length + 2, input.msg.length * 2 + 2);
+
+        assert.equal(JSON.stringify(result), JSON.stringify(expected), true);
+    });
+
+    it("should reveal the payee name correctly", async () => {
+        const witness = await subject(input);
+
+        const expected = Array(textToAsciiArray("<td>Amount:</td>\r\n<td>SGD").length).fill("0")
+            .concat(textToAsciiArray("300.00")).fill("0")
+            .concat(textToAsciiArray("</td>\r\n").fill("0"))
+            .concat(textToAsciiArray("</tr>\r\n<tr>\r\n").fill("0"))
+            .concat(textToAsciiArray("<td>From:</td>\r\n").fill("0"))
+            .concat(textToAsciiArray("<td>PayLah! Wallet (Mobile ending ")).fill("0")
+            .concat(textToAsciiArray("1111")).fill("0")
+            .concat(textToAsciiArray(")</td>\r\n").fill("0"))
+            .concat(textToAsciiArray("</tr>\r\n<tr>\r\n").fill("0"))
+            .concat(textToAsciiArray("<td>To:</td>\r\n<td>")).fill("0")
+            .concat(textToAsciiArray("Steven 'O K-M Jr. "))     // TODO: the space is included in regex extraction
```

Comment:
This is an issue with the regex script. There is an extra space that's extracted after the name. Is there a way to ignore the last ` ` char?